storage/src/test/scala/io/delta/storage/integration/S3LogStoreUtilIntegrationTest.scala

```diff
@@ -0,0 +1,107 @@
+package io.delta.storage.integration
+
+import io.delta.storage.internal.S3LogStoreUtil
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.fs.s3a.{S3AFileSystem, UploadInfo}
+import org.scalatest.Tag
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalactic.source
+
+import java.net.URI
+import java.nio.file.Paths
+import scala.math.max
+import scala.math.ceil
+import scala.math.round
+
+class S3LogStoreUtilIntegrationTest extends AnyFunSuite {
+  private val runIntegrationTests: Boolean =
+    Option(System.getenv("S3_LOG_STORE_UTIL_TEST_ENABLED")).exists(_.toBoolean)
+  private val bucket = System.getenv("S3_LOG_STORE_UTIL_TEST_BUCKET")
+  private val keyPrefix = System.getenv("S3_LOG_STORE_UTIL_TEST_KEY_PREFIX")
+  private lazy val fs: S3AFileSystem = {
+    val fs = new S3AFileSystem()
+    fs.initialize(new URI(s"s3a://$bucket"), configuration)
+    fs
+  }
+  private val configuration = new Configuration()
+  configuration.set( // for local testing only
+    "fs.s3a.aws.credentials.provider",
+    "com.amazonaws.auth.profile.ProfileCredentialsProvider"
+  )
+
+  private val file = Paths.get(getClass.getClassLoader.getResource("sample.json").getPath).toFile
+
+  private def touch(key: String): UploadInfo =
+    fs.putObject(fs.newPutObjectRequest(key, fs.newObjectMetadata(), file))
+
+  private def key(table: String, version: Int): String =
+    s"$keyPrefix/$table/_delta_log/%020d.json".format(version)
+
+  private def path(table: String, version: Int): Path =
+    new Path(s"s3a://$bucket/${key(table, version)}")
+
+  private def version(path: Path): Int
+  = path.getName.takeWhile(_ != '.').toInt
```

Comment:
we have a util for this: `FileNameUtils::deltaVersion`. https://github.com/delta-io/delta/blob/master/storage/src/main/java/io/delta/storage/internal/FileNameUtils.java#L23

Reply:
Thanks! Updated to use this.